frontend/src/components/elements/Audio/Audio.tsx

```diff
@@ -24,12 +24,27 @@ interface Props {
 }
 
 class Audio extends React.PureComponent<Props> {
+  private audioRef = React.createRef<HTMLAudioElement>()
+
+  public componentDidMount = (): void => {
+    if (this.audioRef.current) {
+      const startTime = this.props.element.get("startTime")
+      this.audioRef.current.currentTime = startTime
+    }
```

Comment:
I just tried this locally and realized you need to do the same thing inside `componentDidUpdate` too. Otherwise the start position will not update properly when the app is re-run.

For example, without this fix this app won't work properly:

```python
# This assumes you have an audio file called 'sound.wav')
start = st.slider('Start time', 0, 10, 0, 0.1)
st.audio('sound.wav', start_time=start)
```

---

The fix is something like this:

```python
class Audio extends React.PureComponent<Props> {
  private audioRef = React.createRef<HTMLAudioElement>()

  public componentDidMount = (): void => {
    this.updateTime()
  }

  public componentDidUpdate = (): void => {
    this.updateTime()
  }

  private updateTime(): void {
    if (this.audioRef.current) {
      const startTime = this.props.element.get("startTime")
      this.audioRef.current.currentTime = startTime
    }
  }

  public render(): React.ReactNode {
    const { element, width } = this.props
    const dataUrl =
      "data:" + element.get("format") + ";base64," + element.get("data")
    return (
      <audio
        ref={this.audioRef}
        controls
        src={dataUrl}
        className="stAudio"
        style={{ width }}
      />
    )
  }
}
```

And the same applies for `st.video`.